python/lsst/pipe/tasks/calibrateImage.py

```diff
@@ -880,45 +899,62 @@ def _fit_astrometry(self, exposure, stars):
         result = self.astrometry.run(stars, exposure)
         return result.matches, result.matchMeta
 
-    def _fit_photometry(self, exposure, stars, background):
+    def _fit_photometry(self, exposure, stars):
         """Fit a photometric model to the data and return the reference
         matches used in the fit, and the fitted PhotoCalib.
 
         Parameters
         ----------
         exposure : `lsst.afw.image.Exposure`
             Exposure that is being fit, to get PSF and other metadata from.
-            Modified to be in nanojanksy units, with an assigned photoCalib
-            identically 1.
+            Has the fit `lsst.afw.image.PhotoCalib` attached, with pixel values
+            unchanged.
         stars : `lsst.afw.table.SourceCatalog`
             Good stars selected for use in calibration.
+        background : `lsst.afw.math.BackgroundList`
+            Background model to convert to nanojansky units in place.
 
         Returns
         -------
         calibrated_stars : `lsst.afw.table.SourceCatalog`
             Star catalog with flux/magnitude columns computed from the fitted
-            photoCalib.
+            photoCalib (instFlux columns are retained as well).
         matches : `list` [`lsst.afw.table.ReferenceMatch`]
             Reference/stars matches used in the fit.
-        photoCalib : `lsst.afw.image.PhotoCalib`
+        photo_calib : `lsst.afw.image.PhotoCalib`
             Photometric calibration that was fit to the star catalog.
         """
         result = self.photometry.run(exposure, stars)
         calibrated_stars = result.photoCalib.calibrateCatalog(stars)
-        exposure.maskedImage = result.photoCalib.calibrateImage(exposure.maskedImage)
+        exposure.setPhotoCalib(result.photoCalib)
+        return calibrated_stars, result.matches, result.matchMeta, result.photoCalib
+
+    def _apply_photometry(self, exposure, background):
+        """Apply the photometric model attached to the exposure to the
+        exposure's pixels and an associated background model.
+
+        Parameters
+        ----------
+        exposure : `lsst.afw.image.Exposure`
+            Exposure with the target `lsst.afw.image.PhotoCalib` attached.
+            On return, pixel values will be calibrated and an identity
+            photometric transform will be attached.
+        background : `lsst.afw.math.BackgroundList`
+            Background model to convert to nanojansky units in place.
+        """
+        photo_calib = exposure.getPhotoCalib()
```

Comment:
Why not use `exposure.photoCalib` below, instead of assigning it? Doesn't really matter, I just don't like proliferating the `getX` when we don't have to.

Reply:
Oh, right, good point.